### PR TITLE
Update the driver to use the newest Elasticsearch Java Client API.

### LINKED ISSUE #196: Update ES library to the newest version
Update to use the Elasticsearch newest version: 
https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/introduction.html

### PR DESCRIPTION
**Pull request by dearrudam**
# Minor
- Fix #196
- Made it able to perform mvn test with driver-tck-document without conflict between `javax.json` and `jakarta.json`

### CODE REVIEW

**Comment by dearrudam** [comment]
Hi @otaviojava! Hope you're doing well! Could you review my PR, please?

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

File: jnosql-elasticsearch-driver/pom.xml [hunk 1]
```diff
@@ -42,14 +51,48 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.elasticsearch.client</groupId>
-            <artifactId>elasticsearch-rest-high-level-client</artifactId>
+            <groupId>co.elastic.clients</groupId>
+            <artifactId>elasticsearch-java</artifactId>
             <version>${es.version}</version>
+            <exclusions>
+                <exclusion>
```
Comment (otaviojava) [inline jnosql-elasticsearch-driver/pom.xml:58]: @dearrudam, why did you remove these dependencies?

If it is because of the Jakarta Big bang, please, remove the TCK test that we'll update next year.

We should not change the driver dependencies.

In general, it is an adapter layer.

**Comment by otaviojava** [comment]
@dearrudam please, let me know if I can review/merge it.

File: jnosql-elasticsearch-driver/pom.xml (hunk 1 above)
Comment (dearrudam) [reply jnosql-elasticsearch-driver/pom.xml]: Fixed

**Comment by dearrudam** [comment]
@otaviojava, 

I used the information in this link - [Migrating from the High Level Rest Client to Elasticsearch Java API Client [8.5]](https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/migrate-hlrc.html#_compatibility_mode_using_a_7_17_client_with_elasticsearch_8_x) - in order to make this driver able to works with the newest ES version.

With this PR, the adapter is going to be able to perform custom queries with both APIs: HLRC and Elasticsearch Java API Client [8.5]. 

Could you review it, please?

**Review by otaviojava — APPROVED** [review]
